skrub/_reporting/_table_report.py

```diff
@@ -145,9 +151,14 @@ def _summary_without_plots(self):
 
     @property
     def _any_summary(self):
-        if "_summary_with_plots" in self.__dict__:
-            return self._summary_with_plots
-        return self._summary_without_plots
+        if self.max_plot_columns is None:
+            summary = self._summary_with_plots
+        elif self.max_plot_columns >= self.dataframe.shape[1]:
```

Comment:
```suggestion
        elif self.max_plot_columns >= sbd.shape(self.dataframe)[1]:
```

this change will also require an additional import at the top of the file:

```python
from .. import _dataframe as sbd
```

the way you wrote it with `dataframe.shape` works perfectly, because pandas and polars happen to both expose the `shape` attribute with the same meaning. but to be consistent with the rest of skrub and in case some day we add another backend, for code that could be dealing with either a polars or a pandas dataframe we try to use the backend-agnostic utilities provided in `skrub._dataframe`

Reply:
Yes, that makes sense